modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/ParameterModelParser.java

```diff
@@ -117,4 +118,19 @@ public interface ParameterModelParser extends SemanticTermsParser, AllowedStereo
    * @return a {@link MuleVersion} representing the minimum mule version this component can run on
    */
   Optional<MuleVersion> getMinMuleVersion();
+
+  /**
+   * @return the parameter's {@link InputResolverModelParser} if dynamic metadata were defined
+   */
+  Optional<InputResolverModelParser> getInputResolverModelParser();
+
+  /**
+   * @return the parameter's {@link KeyIdResolverModelParser} if dynamic metadata were defined
+   */
+  Optional<KeyIdResolverModelParser> getKeyIdResolverModelParser(String categoryName);
+
+  /**
+   * @return the order and the existence of a key resolver if metadata key part were defined on the parameter
```

Comment:
The boolean represents if the part is "providedByKeyResolver" ? it is not clear by the docs.